tfe/resource_tfe_project_policy_set.go

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package tfe
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceTFEProjectPolicySet() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceTFEProjectPolicySetCreate,
+		Read:   resourceTFEProjectPolicySetRead,
+		Delete: resourceTFEProjectPolicySetDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceTFEProjectPolicySetImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"policy_set_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceTFEProjectPolicySetCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(ConfiguredClient)
+
+	policySetID := d.Get("policy_set_id").(string)
+	projectID := d.Get("project_id").(string)
+
+	policySetAddProjectsOptions := tfe.PolicySetAddProjectsOptions{}
+	policySetAddProjectsOptions.Projects = append(policySetAddProjectsOptions.Projects, &tfe.Project{ID: projectID})
+
+	err := config.Client.PolicySets.AddProjects(ctx, policySetID, policySetAddProjectsOptions)
+	if err != nil {
+		return fmt.Errorf(
+			"Error attaching policy set id %s to project %s: %w", policySetID, projectID, err)
+	}
+
+	d.SetId(fmt.Sprintf("%s_%s", projectID, policySetID))
+
+	return resourceTFEProjectPolicySetRead(d, meta)
+}
+
+func resourceTFEProjectPolicySetRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(ConfiguredClient)
+
+	policySetID := d.Get("policy_set_id").(string)
+	projectID := d.Get("project_id").(string)
+
+	log.Printf("[DEBUG] Read configuration of project policy set: %s", policySetID)
+	policySet, err := config.Client.PolicySets.ReadWithOptions(ctx, policySetID, &tfe.PolicySetReadOptions{
+		Include: []tfe.PolicySetIncludeOpt{tfe.PolicySetProjects},
+	})
+	if err != nil {
+		if errors.Is(err, tfe.ErrResourceNotFound) {
+			log.Printf("[DEBUG] Policy set %s no longer exists", policySetID)
```

Comment:
@ker-an should we give additional info why the policy set no longer exists? Or is this sufficient?
@laurenolivia

Reply:
Again, if it's ok with @laurenolivia I would do something similar to resource_tfe_workspace_policy_set:

https://github.com/hashicorp/terraform-provider-tfe/blob/03776af85cd657da8b8afb1dfbfb594ff064f7b5/tfe/resource_tfe_workspace_policy_set.go#L74

Reply:
No problem. Not a blocker. Marking this as resolved. Thanks @ker-an and @Netra2104